File: FurnitureStore/FurnitureStore.Models/OrderItem/OrderItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FurnitureStore.Models.OrderItem
{
    public class OrderItem
    {
        public int Id { get; set; }
        public int Quantity { get; set; }
        public int OrderId { get; set; }
        public Order.Order Order { get; set; }
        public long ProductId { get; set; }
        public Product.Product Product { get; set; }
    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/order_item/order_item.dart
import 'package:furniturestore_admin/models/product/product.dart';
import 'package:json_annotation/json_annotation.dart';

part 'order_item.g.dart';

@JsonSerializable()
class OrderItemModel {
  int? id;
  String? productName;
  int? quantity;
  double? price;
  int? orderId;
  int? productId;
  ProductModel? product;

  OrderItemModel({
    this.id,
    this.productName,
    this.quantity,
    this.price,
    this.orderId,
    this.productId,
    this.product,
  });

  factory OrderItemModel.fromJson(Map<String, dynamic> json) {
    var orderItem = _$OrderItemModelFromJson(json);

    if (orderItem.product != null) {
      orderItem.productName ??= orderItem.product!.name;
      orderItem.price ??= orderItem.product!.price;
    }

    return orderItem;
  }

  Map<String, dynamic> toJson() => _$OrderItemModelToJson(this);
}
